### PR TITLE
fix dict like init for ModelOutput

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do?

This PR fixes `ModelOutput` dict-like init as this doesn't work if you don't specify the first attribute. In our case the loss value wasn't cleared. 

###  Reproducer
```python
from transformers.modeling_outputs import CausalLMOutputWithPast
import torch
lm_output = CausalLMOutputWithPast({"logits":torch.ones(2, 3, 4)})
```

before:

```
CausalLMOutputWithPast(loss={'logits': tensor([[[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]],

        [[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]]])}, logits=tensor([[[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]],

        [[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]]]), past_key_values=None, hidden_states=None, attentions=None)
```

after

```
CausalLMOutputWithPast(loss=None, logits=tensor([[[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]],

        [[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [1., 1., 1., 1.]]]), past_key_values=None, hidden_states=None, attentions=None)
````

### CODE REVIEW

**Review by Cyrilvallez — APPROVED** [review]
Makes sense! Thanks!
